sugarscape.py

```diff
@@ -154,24 +154,35 @@ def configureDiseases(self, numDiseases):
 
         diseaseEndowments = self.randomizeDiseaseEndowments(numDiseases)
         random.shuffle(self.agents)
-        diseases = []
         for i in range(numDiseases):
             diseaseID = self.generateDiseaseID()
             diseaseConfiguration = diseaseEndowments[i]
             newDisease = disease.Disease(diseaseID, diseaseConfiguration)
-            diseases.append(newDisease)
+            self.diseases.append(newDisease)
+        startingDiseases = self.configuration["startingDiseasesPerAgent"]
 
-        unplacedDisease = 0
-        for agent in self.agents:
-            for newDisease in diseases:
-                hammingDistance = agent.findNearestHammingDistanceInDisease(newDisease)["distance"]
-                if hammingDistance != 0:
+        if startingDiseases == [0, 0]:
+            diseases = self.diseases[:]
+            for agent in self.agents:
+                for newDisease in diseases:
+                    hammingDistance = agent.findNearestHammingDistanceInDisease(newDisease)["distance"]
+                    if hammingDistance != 0:
+                        agent.catchDisease(newDisease)
+                        diseases.remove(newDisease)
+                        break
+            if len(diseases) > 0 and ("all" in self.debug or "sugarscape" in self.debug):
+                print(f"Could not place {len(diseases)} diseases.")
+        else:
+            minStartingDiseases = startingDiseases[0]
+            maxStartingDiseases = startingDiseases[1]
+            currStartingDiseases = minStartingDiseases
+            for agent in self.agents:
+                agentDiseases = random.sample(self.diseases, currStartingDiseases)
+                for newDisease in agentDiseases:
                     agent.catchDisease(newDisease)
-                    self.diseases.append(newDisease)
-                    diseases.remove(newDisease)
-                    break
-        if len(diseases) > 0 and ("all" in self.debug or "sugarscape" in self.debug):
```

Comment:
Some form of error message needs to remain for both cases ([0,0] for starting diseases and the new base case) to warn users if their desired number of diseases aren't able to be placed.

Reply:
The new logic gives every agent a certain number of diseases, so the only issue would be that the program is attempting to give more unique diseases to an agent than exist in the master list. This would be a better fit for `verifyConfiguration`, so I'll add it there.